optd-persistent/src/lib.rs

```diff
@@ -1,11 +1,49 @@
+#![allow(dead_code)]
+
+use std::sync::atomic::AtomicUsize;
+
 use sea_orm::*;
 use sea_orm_migration::prelude::*;
 
-mod migrator;
 use migrator::Migrator;
 
+mod entities;
+mod migrator;
+
+mod cost_model;
+pub use cost_model::interface::CostModelStorageLayer;
+
+pub type CostModelStorageResult<T> = Result<T, CostModelError>;
+
+pub enum CostModelError {
+    // TODO: Add more error types
+    Database(DbErr),
+}
+
+impl From<DbErr> for CostModelError {
+    fn from(value: DbErr) -> Self {
+        CostModelError::Database(value)
+    }
+}
+
+pub struct BackendManager {
+    db: DatabaseConnection,
+    latest_epoch_id: AtomicUsize,
+}
+
+impl BackendManager {
+    /// Creates a new `BackendManager`.
+    pub async fn new() -> CostModelStorageResult<Self> {
+        Ok(Self {
+            db: Database::connect(DATABASE_URL).await?,
+            latest_epoch_id: AtomicUsize::new(0),
+        })
+    }
+}
```

Comment:
Can you create a new error called `BackendError` that has a single variant `CostModel` so that it is easy to integrate the `MemoError` from my branch?

Reply:
What do you mean? A enum?
```
pub enum BackendError {
    CostModel(CostModelError),
}

impl From<CostModelError> for BackendError {
    fn from(value: CostModelError) -> Self {
        BackendError::CostModel(value)
    }
}

Reply:
Yes and the `new` function should return a `Result` where `type Result<T> = std::result::Result<T, BackendError>`

Reply:
Updated!